libutils/known_dirs.c

```diff
@@ -159,6 +159,23 @@ const char *GetBinDir(void)
     }
 }
 
+const char *GetKeyDir(bool *allocated)
+{
+    const char *keydir = getenv("CFENGINE_TEST_OVERRIDE_KEY_DIR");
+
+    if (keydir == NULL) 
+    {
+        const char* workdir = GetWorkDir(); 
+        char dest[PATH_MAX];
+        snprintf(dest, PATH_MAX, "%s" FILE_SEPARATOR_STR "%s", workdir, "ppkeys");
+        (*allocated) = true;
+
+        return strdup(dest);
+    }
+    (*allocated) = false;
+    return keydir;
+}
```

Comment:
@victormlg Why is this one so different from all the other ones?

All the other similar functions seem to use `GET_DEFAULT_DIRECTORY_DEFINE` macro and do not have a `allocated` parameter.

Reply:
I think if you do it the "as-is way" you may need to adjust configure.ac as we provide configuration options to change the location of the various directories.